custom_model_runner/datarobot_drum/drum/custom_tasks/fit_adapters/python/python_fit_adapter.py

```diff
@@ -0,0 +1,132 @@
+All rights reserved.
+This is proprietary source code of DataRobot, Inc. and its affiliates.
+Released under the terms of DataRobot Tool and Utility Agreement.
+"""
+import json
+import logging
+import sys
+
+import pandas as pd
+
+from datarobot_drum.drum.custom_tasks.fit_adapters.base_fit_adapter import BaseFitAdapter
+from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.model_adapter import PythonModelAdapter
+from datarobot_drum.drum.utils import make_sure_artifact_is_small, StructuredInputReadUtils
+
+logger = logging.getLogger(LOGGER_NAME_PREFIX + "." + __name__)
+
+
+class PythonFitAdapter(BaseFitAdapter):
+    def configure(self):
+        super(PythonFitAdapter, self).configure()
+        sys.path.append(self.custom_task_folder_path)
+
+    def _extract_weights(self, X):
+        # extract weights from file or data
+        if self.weights_filename:
+            row_weights = pd.read_csv(self.weights_filename).sample(self.num_rows, random_state=1)
+        elif self.weights:
+            if self.weights not in X.columns:
+                raise ValueError(
+                    "The column name {} is not one of the columns in "
+                    "your training data".format(self.weights)
+                )
+            row_weights = X[self.weights]
+        else:
+            row_weights = None
+        return row_weights
+
+    def _extract_class_order(self):
+        # get class order obj from class labels
+        if self.negative_class_label is not None and self.positive_class_label is not None:
+            class_order = [self.negative_class_label, self.positive_class_label]
+        elif self.class_labels:
+            class_order = self.class_labels
+        else:
+            class_order = None
+
+        return class_order
+
+    def _python_fit_preprocessing(self):
+        """
+        Shared preprocessing to get X, y, class_order, row_weights, and parameters.
+        Used by _materialize method for both python and R fitting.
+
+        :param fit_class: PythonFit or RFit class
+        :return:
+            X: pd.DataFrame of features to use in fit
+            y: pd.Series of target to use in fit
+            class_order: array specifying class order, or None
+            row_weights: pd.Series of row weights, or None
+        """
+        df = StructuredInputReadUtils.read_structured_input_file_as_df(
+            filename=self.input_filename, sparse_column_file=self.sparse_column_filename
+        )
+
+        # get num rows to use
+        if self.num_rows == "ALL":
+            self.num_rows = len(df)
+        else:
+            if self.num_rows > len(df):
+                raise DrumCommonException(
+                    "Requested number of rows greater than data length {} > {}".format(
+                        self.num_rows, len(df)
+                    )
+                )
+            self.num_rows = int(self.num_rows)
```

Comment:
shouldn't we cast to int before checking?

Reply:
probably. It's a safeguard to make sure it's not float. A float is fine for the comparison above